core/node/consistency_checker/src/lib.rs

```diff
@@ -829,3 +806,34 @@ async fn wait_for_l1_batch_with_metadata(
             .ok();
     }
 }
+
+// Get settlement layer based on ETH tx, all eth txs should be presented on settlement layer. what is the best place for this function?
+pub async fn get_settlement_mode(
```

Comment:
we save chain id for eth txs (or if chain_id is missing means it's old server and thus it's l1), I believe it should be used to determine settlement mode.

Reply:
oh, nice thank you! didn't know about it